### DISCUSSION TITLE
Dynamically Add Elements to stencil

### DISCUSSION THREAD

**Discussion by RajeshNakkanaboina** [clientIO/joint#2226] [Q&A] [answered]
### Introduction

I have a requirement of adding elements dynamically(programmatically) to stencil. In which the product information coming from external API, Can anyone know how can we achieve above scenario. 

Thanks

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
Here's the example: https://www.jointjs.com/demos/drop-stencil-element-as-shape-icon

**Reply by RajeshNakkanaboina** [reply]
Hello, kumilingus Regards for the response
Want to dynamically (programmatically) add elements, goods and shapes to the kitchen sink model rather than loading a static stencil. Would you kindly explain to me how to add for kitchen sink?

**Reply by kumilingus** [reply]
As shown in the example, you can load the stencil elements at any point using the `stencil.load()` method.

If you want to use stencil groups, you should follow these 3 steps:
```js
// define the stencil groups
stencil.options.groups = { a: { label: 'A' }, b: { label: 'B' }}
// render the groups
stencil.render();
// load the stencil shapes
stencil.load({
   a: [new joint.shapes.standard.Rectangle({ size: { width: 100, height: 100 }})],
   b: [new joint.shapes.standard.Circle({ size: { width: 100, height: 100 }})]
});

**Comment by RajeshNakkanaboina** [comment]
Hi kumilingus,
Thanks for your reply.  I'm attempting to dynamically add objects or shapes to a stencil using a button click in JointJ's kitchen sink app. I expected a shape to be added when I clicked the button, but the stencil didn't have that shape. when we add the shape, how to reload the stencil. 
![Screenshot (104)](https://github.com/clientIO/joint/assets/134581112/3df45a0f-a084-4a65-8900-ccafbd4ae893)
![Screenshot (105)](https://github.com/clientIO/joint/assets/134581112/bdef8a99-7ae4-4612-9ff2-f8567c9084f5)

Verify the screenshots in the section above. After pressing a button, it changed from 29 to 30 shapes initially. What should I do to reload the stencil after adding the product, if you could possibly advise me?

**Reply by kumilingus** [reply]
Here's an example of how to add elements to the stencil dynamically: https://www.jointjs.com/demos/dynamic-stencil

You need to reload the stencil with `stencil.load()`.

**Reply by RajeshNakkanaboina** [reply]
I don't want to load dynamic stencil. I want to load shapes/products dynamically in kitchen sink app. stencil.load() is also not working
Could you please guide me how to do in kitchen sink. Is there any methods??
![Screenshot (110)](https://github.com/clientIO/joint/assets/134581112/ffedf6ee-f883-4aa7-a042-47b96e5a9bd5)
![Screenshot (109)](https://github.com/clientIO/joint/assets/134581112/29da357c-3016-4d30-89a8-9c50e29e2e97)
Please find the above attachments

**Reply by kumilingus** [reply]
The example I have shared shows how to add a new element to the existing stencil dynamically. That is what you are looking for, correct?

`App.config.stencil` is just a JSON with stencil configuration.
`app.stencil` is the instance of the stencil UI plugin on left-hand side.

**Reply by RajeshNakkanaboina** [reply, marked as answer]
Yes, Kumilingus
will check with app.stencil.load()

Thanks for the reply
